src/main/scala/sigmastate/serialization/MethodCallSerializer.scala

```diff
@@ -17,25 +17,38 @@ case class MethodCallSerializer(opCode: Byte, cons: (Value[SType], SMethod, Inde
       assert(mc.args.nonEmpty)
       w.putValues(mc.args)
     }
-    w.putUByte(mc.typeSubst.size)
-    mc.typeSubst.foreach { case (typeIdent, tpe) => w.putType(typeIdent).putType(tpe) }
+    mc.method.stype match {
+      case genType: SGenericType =>
+        w.putUByte(mc.typeSubst.size)
+        mc.typeSubst.foreach { case (ti, tpe) =>
+          val tpIndex = genType.substitutedTypeParams.indexOf(STypeParam(ti))
+          w.putUByte(tpIndex)
```

Comment:
@greenhat we don't need to output tpIndex either. We just need to always serialize them is the order specified in SGenericType.typeParams

Reply:
@aslesarenko Damn, that's so obvious in the hindsight! Thanks. This also catches missed type substitutions. Done in 
https://github.com/ScorexFoundation/sigmastate-interpreter/pull/400/commits/eb4b3c177c7db1583329e177c030d971987af40f